test/11_array.jl

```diff
@@ -188,8 +188,16 @@ using CategoricalArrays: DefaultRefType, leveltype
     @test x[3] === CategoricalValue(x.pool, 1)
 
     @test_throws ArgumentError levels!(x, ["a"])
+    # check that x is restored correctly when dropping levels is not allowed
+    @test x == ["a", "b", "b"]
+    @test levels(x) == ["b", "a"]
+
     @test_throws ArgumentError levels!(x, ["e", "b"])
+
     @test_throws ArgumentError levels!(x, ["e", "a", "b", "a"])
+    # one again check that x is restored correctly when dropping levels is not allowed
+    @test x == ["a", "b", "b"]
+    @test levels(x) == ["b", "a"]
```

Comment:
Sorry to be a pain, but there are equivalent lines in 12_missingarray.jl: can you keep them in sync? :-)

Reply:
No problem :) Done